src/Jobs/BaseJob.php

```diff
@@ -387,13 +387,13 @@ public function retry(string $group, string $message, int $delay = 0): int {
      *
      * @throws LostLockException
      */
-    public function heartbeat(array $data = []): float
+    public function heartbeat(?array $data = null): float
     {
         try {
             $this->expires = $this->client->heartbeat(
                 $this->jid,
                 $this->worker,
-                json_encode($data, JSON_UNESCAPED_SLASHES)
+                json_encode(\is_array($data) ? $data : $this->data, JSON_UNESCAPED_SLASHES)
```

Comment:
It will be better to use `$data !== null` instead of `is_array($data)`. `is_array` is marginally slower due to function call overhead.

Reply:
Marginally being the operative word there. As of at least 7.0 `is_array` checks are optimised into a ZEND_TYPE_CHECK opcode.

To demonstrate how this affects the program I ran a little test.
Calling `is_array` vs `!== null` in a loop 100M times (and running the tests 5 or 6 times to compare the differences), I see about 0.02 to 0.03 seconds difference... across one hundred million invocations.

If you want to claim that the project has a preference for comparison operators over is_* functions for stylistic reasons, I won't necessarily agree with you but it's easily something that can be defined and checked by a code style linter.

Asking for it because you think it's faster, is not something I'm going to agree on, and this type of claim is kind of silly anyway, given that the result of said call is about to be sent over a network connection.